frb_codegen/src/library/utils/test_utils.rs

```diff
@@ -28,28 +28,43 @@ pub(crate) fn json_golden_test(
     let actual: Value = serde_json::from_str(&actual_str)?;
     debug!("json_golden_test sanitizers={sanitizers:?} actual:\n{actual_str}");
 
-    raw_golden_test(actual, &actual_str, matcher_path, |x| {
-        Ok(serde_json::from_str(&x)?)
-    })
+    raw_golden_test(
+        actual,
+        &actual_str,
+        matcher_path,
+        |x| Ok(serde_json::from_str(&x)?),
+        None,
+    )
 }
 
 pub(crate) fn text_golden_test(actual: String, matcher_path: &Path) -> anyhow::Result<()> {
-    raw_golden_test(actual.clone(), &actual, matcher_path, |x| {
+    raw_golden_test(
+        actual.clone(),
+        &actual,
+        matcher_path,
         // Otherwise tests in macos/linux passes but fails on windows
-        Ok(x.replace("\r\n", "\n"))
-    })
+        |x| Ok(x.replace("\r\n", "\n")),
+        Some(|expect, actual| {
+            #[cfg(test)]
+            use pretty_assertions::assert_str_eq as assert_eq;
+            assert_eq!(expect, actual);
+        }),
+    )
 }
 
 fn raw_golden_test<T, F>(
     actual: T,
     actual_str: &str,
     matcher_path: &Path,
     deserializer: F,
+    asserter: Option<fn(&T, &T)>,
 ) -> anyhow::Result<()>
 where
     T: Eq + Debug,
     F: Fn(String) -> anyhow::Result<T>,
 {
+    #[cfg(test)]
```

Comment:
nit: seems that we can make the whole fn or mod be `cfg(text)` since it is only used for tests